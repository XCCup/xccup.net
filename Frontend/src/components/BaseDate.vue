<template>
  {{ format(new Date(timestamp), dateFormat) }}
</template>

<script>
import { format } from "date-fns";
export default {
  name: "BaseDate",
  data() {
    return {
      format,
    };
  },
  props: {
    timestamp: {
      type: [String],
      required: true,
    },
    dateFormat: {
      type: String,
      default: "dd.MM.yyyy",
    },
  },
};
</script>
