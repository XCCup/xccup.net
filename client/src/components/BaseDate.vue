<template>
  <span v-if="timestamp">{{ format(new Date(timestamp), dateFormat) }}</span>
</template>

<script setup>
import { format } from "date-fns";
defineProps({
  timestamp: {
    type: [String, Number, Date],
    required: true,
  },
  dateFormat: {
    type: String,
    default: "dd.MM.yyyy",
  },
});
</script>
