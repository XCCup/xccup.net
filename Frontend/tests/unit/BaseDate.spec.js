import { mount } from "@vue/test-utils";
import BaseDate from "@/components/BaseDate";

test("Takes a timestamp and a configuration parameter and renders a date", () => {
  const timestamp = 1621088955000;
  const dateFormat = "dd.MM.yyyy";
  const wrapper = mount(BaseDate, { props: { timestamp, dateFormat } });

  expect(wrapper.html()).toMatchSnapshot();
});

test("Use a default date format if none is passed", () => {
  const timestamp = 1621088955000;
  const wrapper = mount(BaseDate, { props: { timestamp } });

  expect(wrapper.html()).toMatchSnapshot();
});

test("Do not render if no timestamp prop is passed", () => {
  const timestamp = null;
  const wrapper = mount(BaseDate, { props: { timestamp } });

  expect(wrapper.html()).toMatchSnapshot();
});
