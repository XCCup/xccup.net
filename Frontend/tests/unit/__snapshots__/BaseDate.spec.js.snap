// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Do not render if no timestamp prop is passed 1`] = `<!--v-if-->`;

exports[`Takes a timestamp and a configuration parameter and renders a date 1`] = `<span>15.05.2021</span>`;

exports[`Use a default date format if none is passed 1`] = `<span>15.05.2021</span>`;
